### ISSUE TITLE
IPv6 Support causes existing module to fail

### ISSUE THREAD

**Issue by eratio08** [Guimove/terraform-aws-bastion#129]
The commit https://github.com/Guimove/terraform-aws-bastion/commit/df6830573f2832e9a50e419c8284c133fda19463 causes the following error if the new module version is used with existing resources of the module.

```shell
│ Error: Unsupported attribute
│ 
│   on .terraform/modules/bastion/main.tf line 88, in resource "aws_security_group_rule" "ingress_bastion":
│   88:   ipv6_cidr_blocks = concat(data.aws_subnet.subnets.*.ipv6_cidr_blocks, var.ipv6_cidrs)
│ 
│ This object does not have an attribute named "ipv6_cidr_blocks".
╵
╷
│ Error: Unsupported attribute
│ 
│   on .terraform/modules/bastion/main.tf line 88, in resource "aws_security_group_rule" "ingress_bastion":
│   88:   ipv6_cidr_blocks = concat(data.aws_subnet.subnets.*.ipv6_cidr_blocks, var.ipv6_cidrs)
│ 
│ This object does not have an attribute named "ipv6_cidr_blocks".
╵
╷
│ Error: Unsupported attribute
│ 
│   on .terraform/modules/bastion/main.tf line 88, in resource "aws_security_group_rule" "ingress_bastion":
│   88:   ipv6_cidr_blocks = concat(data.aws_subnet.subnets.*.ipv6_cidr_blocks, var.ipv6_cidrs)
│ 
│ This object does not have an attribute named "ipv6_cidr_blocks".
```
It seems to be caused by subnets that do not use ipv6.

**Comment by neurogamer** [comment]
Same issue. We need 2.3.0

**Comment by Guimove** [comment]
Please check with the version  v2.3.2 (https://github.com/Guimove/terraform-aws-bastion/releases/tag/v2.3.2)

**Comment by eratio08** [comment]
v2.3.2 causes a different issue now

```
╷
│ Error: "" is not a valid CIDR block: invalid CIDR address: 
│ 
│   with module.bastion.aws_security_group_rule.ingress_bastion[0],
│   on .terraform/modules/bastion/main.tf line 88, in resource "aws_security_group_rule" "ingress_bastion":
│   88:   ipv6_cidr_blocks = concat(data.aws_subnet.subnets.*.ipv6_cidr_block, var.ipv6_cidrs)
│ 
╵
╷
│ Error: "" is not a valid CIDR block: invalid CIDR address: 
│ 
│   with module.bastion.aws_security_group_rule.ingress_bastion[0],
│   on .terraform/modules/bastion/main.tf line 88, in resource "aws_security_group_rule" "ingress_bastion":
│   88:   ipv6_cidr_blocks = concat(data.aws_subnet.subnets.*.ipv6_cidr_block, var.ipv6_cidrs)
│ 
╵
╷
│ Error: "" is not a valid CIDR block: invalid CIDR address: 
│ 
│   with module.bastion.aws_security_group_rule.ingress_bastion[0],
│   on .terraform/modules/bastion/main.tf line 88, in resource "aws_security_group_rule" "ingress_bastion":
│   88:   ipv6_cidr_blocks = concat(data.aws_subnet.subnets.*.ipv6_cidr_block, var.ipv6_cidrs)
```

**Comment by Guimove** [comment]
Ok thanks for the notice, I didn't get this error on my test. 

For the moment, please stick to the version 2.3.0

I'll try to take time to look at this asap.

**Comment by alsidneio** [comment]
The `ipv6_cidrs`  throws an error on the `aws_security_group_rule`  resource if the `aws_subnet ` resource does not have ipv6 cidr block.  The `ipv6_cidr_blocks` argument needs to be set to null if the concat function returns an empty string

**Comment by alsidneio** [comment]
PR to fix ipv6 errors: https://github.com/Guimove/terraform-aws-bastion/pull/132

**Comment by Guimove** [comment]
It's resolved in [v3.0.1](https://github.com/Guimove/terraform-aws-bastion/tree/v3.0.1)

Thanks for your support @alsidneio
